src/Sulu/Bundle/AdminBundle/Resources/js/components/CircleSelection/CircleSelection.js

```diff
@@ -1,88 +1,233 @@
+                new SizeNormalizer(
+                    containerWidth,
+                    containerHeight,
+                    maxRadius,
+                    minRadius
+                )
+            );
+        }
+
+        if (round) {
+            normalizers.push(new RoundingNormalizer());
+        }
+
+        return normalizers;
+    }
+
+    @computed get normalizers() {
+        return CircleSelectionComponent.createNormalizers(this.props);
+    }
+
+    normalize(selection: SelectionData): SelectionData {
+        return this.normalizers.reduce((data, normalizer) => normalizer.normalize(data), selection);
+    }
+
+    @computed get maximumSelection(): SelectionData {
+        const {containerWidth, containerHeight, resizable, value} = this.props;
+
+        const radius = resizable
+            ? Math.min(containerWidth, containerHeight) / 2
+            : (value && value.radius) || 0;
+
+        return this.normalize(
+            this.centerSelection({
+                left: 0,
+                top: 0,
+                radius,
+            })
+        );
+    }
+
+    centerSelection(selection: SelectionData): SelectionData {
+        const {containerWidth, containerHeight} = this.props;
+
+        const halfWidth = containerWidth / 2;
+        const halfHeight = containerHeight / 2;
+
+        return {
+            ...selection,
+            left: halfWidth,
+            top: halfHeight,
+        };
+    }
+
+    handleCircleDoubleClick = () => {
+        const {onChange, resizable} = this.props;
+
+        if (resizable) {
+            onChange(this.maximumSelection);
+
+            return;
+        }
+
+        onChange(this.normalize(this.centerSelection(this.value)));
+    };
+
+    handleCircleChange = (value: SelectionData) => {
+        const {onChange} = this.props;
+
+        onChange(this.normalize(value));
+    };
+
     render() {
-        const {
-            children,
-            containerHeight,
-            containerWidth,
-            disabled,
-            label,
-            maxRadius,
-            minRadius,
-            onChange,
-            resizable,
-            round,
-            skin,
-            usePercentageValues,
-            value,
-        } = this.props;
+        const {disabled, label, onFinish, resizable, skin} = this.props;
+        const {left, top, radius} = this.value;
+
+        return (
+            <ModifiableCircle
+                disabled={disabled}
+                label={label}
+                left={left}
+                onChange={this.handleCircleChange}
+                onDoubleClick={this.handleCircleDoubleClick}
+                onFinish={onFinish}
+                radius={radius}
+                resizable={resizable}
+                skin={skin}
+                top={top}
+            />
+        );
+    }
+}
+
+const CircleSelectionWrapper = withPercentageValues(CircleSelectionComponent);
+
+class CircleSelectionContainer extends React.Component<Props> {
```

Comment:
`Container` is a bit misleading here, because in our react architecture `containers` are somethings different... Can't we just name this one `CircleSelection`?

Reply:
Wait until this discussion is resolved: https://github.com/sulu/sulu/pull/5434#discussion_r498299726